include/proxy-wasm/wasm.h

```diff
@@ -357,7 +357,7 @@ class PluginHandleBase : public std::enable_shared_from_this<PluginHandleBase> {
                             std::shared_ptr<PluginBase> plugin)
       : plugin_(plugin), wasm_handle_(wasm_handle) {}
   ~PluginHandleBase() {
-    if (wasm_handle_) {
+    if (wasm_handle_ && wasm_handle_->wasm() && plugin_) {
       wasm_handle_->wasm()->startShutdown(plugin_->key());
```

Comment:
I have mixed feelings about this. On one hand, it avoids `nullptr`-dereference, but it also hides that the fact that `PluginHandleBase` is completely broken (it's only job is to start the shutdown sequence once the handle is destroyed, and it won't be doing that if either of those are missing).

How do you end up in this state?